services/QuillLMS/app/workers/set_impact_metrics_worker.rb

```diff
@@ -3,33 +3,30 @@
 class SetImpactMetricsWorker
   include Sidekiq::Worker
 
-  ACTIVITY_SESSION_MINIMUM = 9
   FREE_LUNCH_MINIMUM = 39
 
   def perform
     set_impact_metrics
   end
 
   def set_impact_metrics
-    finished_activity_sessions_query = ActivitySession.unscoped.where(state: 'finished')
-    teachers_query = User.select(:id)
-      .joins(:units, classroom_units: :activity_sessions)
-      .where("activity_sessions.state = 'finished'")
-      .group('users.id')
-      .having('count(activity_sessions) > ?', ACTIVITY_SESSION_MINIMUM)
-    teacher_ids = teachers_query.to_a.map(&:id)
-    schools_query = School.joins(:schools_users).where(schools_users: {user_id: teacher_ids}).distinct
-    low_income_schools_query = schools_query.where("free_lunches > ?", FREE_LUNCH_MINIMUM)
+    finished_activity_sessions_query = QuillBigQuery::ActivitiesAllTimeQuery.run
+
+    teachers_query = QuillBigQuery::ActiveTeachersAllTimeQuery.run
+    teacher_ids = teachers_query.to_a.map {|teacher| teacher["id"]}
+
+    schools_query = QuillBigQuery::SchoolsContainingCertainTeachersQuery.run(teacher_ids)
+    low_income_schools_query = schools_query.select { |school| school["free_lunches"].present? && school["free_lunches"] > FREE_LUNCH_MINIMUM}
 
     number_of_sentences = finished_activity_sessions_query.count.floor(-5) * 10
```

Comment:
Nit, since this was here before: abstract the magical arithmetic `floor(-5) * 10` into a helper method